internal/plugins/helm/api.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package helm
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+
+	gencrd "github.com/operator-framework/operator-sdk/internal/generate/crd"
+	sdkscaffold "github.com/operator-framework/operator-sdk/internal/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/helm"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/input"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+)
+
+type createAPIPlugin struct {
+	config *config.Config
+
+	// For help text.
+	commandName string
+
+	// Helm flags
+	APIVersion       string
+	Kind             string
+	CRDVersion       string
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
+}
+
+var (
+	_ plugin.CreateAPI = &createAPIPlugin{}
+)
+
+func (p *createAPIPlugin) UpdateContext(ctx *plugin.Context) {
+	ctx.Description = `Create a Kubernetes API by creating a CR and CRD with the Helm Chart package directories.`
+	p.commandName = ctx.CommandName
+}
+
+func (p *createAPIPlugin) BindFlags(fs *pflag.FlagSet) {
+	const (
+		apiVersion       = "api-version"
+		kind             = "kind"
+		crdVersion       = "crd-version"
+		helmChart        = "helm-chart"
+		helmChartVersion = "helm-chart-version"
+		helmChartRepo    = "helm-chart-repo"
+	)
+
+	fs.StringVar(&p.APIVersion, "api-version", "",
+		"Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)")
+	fs.StringVar(&p.Kind, kind, "", "Kubernetes resource Kind name. (e.g AppService)")
+	fs.StringVar(&p.CRDVersion, "crd-version", gencrd.DefaultCRDVersion,
+		"CRD version to generate")
+	fs.StringVar(&p.HelmChartRef, helmChart, "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path).")
+	fs.StringVar(&p.HelmChartVersion, helmChartVersion, "",
+		"Specific version of the helm chart (default is latest version)")
+	fs.StringVar(&p.HelmChartRepo, helmChartRepo, "",
+		"Chart repository URL for the requested helm chart")
+}
+
+func (p *createAPIPlugin) InjectConfig(c *config.Config) {
+	c.Layout = plugin.KeyFor(Plugin{})
```

Comment:
Only `Init` should do this.
```suggestion
```
There should probably be a follow-up in kb to do this automatically, otherwise every `Init` plugin must set the layout themselves.

Reply:
good catcher.